collection_json.py

```diff
@@ -6,6 +6,59 @@
 __version__ = '0.1.0'
 
 
+class ArrayProperty(object):
+
+    """Property which converts from any enumerable to a typed Array instance."""
+
+    def __init__(self, cls, name):
+        """Constructs typed array property
+
+        :param cls type: the type of objects expected in the array
+        :param name str: name of the property
+        """
+        self.cls = cls
+        self.name = name
+        self.value_attr = "_" + name
```

Comment:
We are close. There is a special case here that would make this work:
```
If an object defines both __get__() and __set__(), it is considered a data descriptor. Descriptors that only define __get__() are called non-data descriptors (they are typically used for methods but other uses are possible).

Data and non-data descriptors differ in how overrides are calculated with respect to entries in an instance’s dictionary. If an instance’s dictionary has an entry with the same name as a data descriptor, the data descriptor takes precedence. If an instance’s dictionary has an entry with the same name as a non-data descriptor, the dictionary entry takes precedence.
```

Since we implement both get and set, the data descriptor will always take precedence. So we can use the same name as passed in. But let me be clear, instance.__dict__[name] is just a more explicit way of getattr, setattr, so its just an equivalent way of doing what I was doing. In fact the __dict__ approach, it would fail if we decided to use, say, slot classes for your objects (something totally reasonable as a future enhancement). 

Knowing this, I'm going to implement current form with getattr, setattr -- but throwing an attribute error if the getattr doing work, and using the passed in name as the dictionary entry.